contract/src/msgs/owner/execute/pause.rs

```diff
@@ -0,0 +1,25 @@
+use cosmwasm_std::{DepsMut, Env, Event, MessageInfo, Response};
+use seda_common::msgs::owner::execute;
+
+use crate::{
+    contract::CONTRACT_VERSION,
+    error::ContractError,
+    msgs::{owner::state::OWNER, ExecuteHandler},
+    state::PAUSED,
+};
+
+impl ExecuteHandler for execute::pause::Execute {
+    fn execute(self, deps: DepsMut, _: Env, info: MessageInfo) -> Result<Response, ContractError> {
+        // require the sender to be the OWNER
+        let owner = OWNER.load(deps.storage)?;
+        if info.sender != owner {
+            return Err(ContractError::NotOwner);
+        }
+
+        PAUSED.save(deps.storage, &true)?;
+
+        Ok(Response::new().add_events([
```

Comment:
Maybe add an attribute to say if it's paused or not?

Reply:
Otherwise the events for pause and unpause are the same.
Should we allow to pause an already paused contract and throw an event? :P

Reply:
It's a good question. We could error instead saying paused/unpaused.

Or we could do a different event saying attempted to pause/unpause?

Reply:
I'm indifferent to whether we silently do nothing or throw an error 😄 We can use the same event, just add a boolean (string) attribute `"paused": "true|false"`